File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/features/HandFeature.java
package org.firstinspires.ftc.teamcode.features;

import org.firstinspires.ftc.teamcode.internals.features.Buildable;
import org.firstinspires.ftc.teamcode.internals.features.Feature;
import org.firstinspires.ftc.teamcode.internals.hardware.Devices;
import org.firstinspires.ftc.teamcode.internals.misc.Clock;

public class HandFeature extends Feature implements Buildable {

    private boolean open = true;

    @Override
    public void build() {
        Devices.servo0.setPosition(50);
        Devices.servo1.setPosition(50);
        Clock.make("hand");
    }

    @Override
    public void loop() {
        if(open && Devices.distanceSensor.getDistance() < 33 && Clock.get("hand").elapsed(2)) {
            Devices.servo0.setPosition(88);
            Devices.servo1.setPosition(12);
            open = false;
        }else if(!open && Devices.controller1.getA()) {
            Devices.servo0.setPosition(50);
            Devices.servo1.setPosition(50);
            open = true;
            Clock.get("hand").reset();
        }
    }

}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/features/HandFeature.kt
package org.firstinspires.ftc.teamcode.features

import com.acmerobotics.roadrunner.util.NanoClock
import org.firstinspires.ftc.teamcode.internals.features.Buildable
import org.firstinspires.ftc.teamcode.internals.features.Feature
import org.firstinspires.ftc.teamcode.internals.hardware.Devices

class HandFeature : Feature(), Buildable {
    private var open = true
    private var second = 0.0
    override fun build() {
        Devices.servo0.position = 50.0
        Devices.servo1.position = 50.0
    }

    override fun loop() {
        /*
        Pressing x on either controller releases the hand
         */
        if (open && Devices.distanceSensor.distance < 33 && NanoClock.system().seconds() > second) {
            Devices.servo0.position = 88.0
            Devices.servo1.position = 12.0
            open = false
        } else if (!open && Devices.controller1.a) {
            Devices.servo0.position = 50.0
            Devices.servo1.position = 50.0
            open = true
            second = NanoClock.system().seconds() + 2
        }
    }
}